### PR TITLE
fix: shell quoting on playwright flags

### PR DESCRIPTION
**Pull request by ferrandiaz**
## Affected Components
* [x] CLI
* [ ] Create CLI
* [ ] Test
* [ ] Docs
* [ ] Examples
* [ ] Other

### Summary
- Add shared shell quoting utility (`src/services/shell.ts`)
- Fix argument quoting in `pw-test`, `buildTestCommand`, and package manager
- Remove duplicated quoting logic
### Problem
Shell commands were constructed with inconsistent/incomplete quoting:
- `pw-test` only quoted args containing spaces, missing other metacharacters (`|`, `$`, `;`, `*`, etc.)
- `buildTestCommand` had manual quoting that missed edge cases
- `package-manager` had a local `unsafeQuoteArg` function not shared elsewhere
### Solution
Created a shared `shell.ts` utility with consistent quoting that handles all shell metacharacters. Safe characters (`%+,-./:=@_` and alphanumeric) pass through unquoted; everything else gets properly escaped.

### CODE REVIEW

File: packages/cli/src/services/shell.ts [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+  if (arg === '') {
+    return '\'\''
+  }
+  if (SAFE_SHELL_CHARS.test(arg)) {
+    return arg
+  }
+  return `'${arg.replace(/'/g, '\'"\'"\'')}'`
+}
+
+export function shellQuoteDouble (arg: string): string {
```
Comment (sorccu) [inline packages/cli/src/services/shell.ts]: Are you sure we want a function that does double quotes? While these are not meant to be perfectly safe (as the previous name I used indicated), double quotes are more trouble because they allow things like variable interpolation. You'd have to cover at least `$` and possibly a few more characters to make it safer (though never perfectly safe). It's just a lot of trouble. Why not use single quotes everywhere?

**Review by sorccu — CHANGES_REQUESTED** [review]
See my comment.

File: packages/cli/src/services/shell.ts [hunk 2]
```diff
@@ -0,0 +1,26 @@
+// Safe shell characters: %+,-./:=@_ and alphanumeric
+const SAFE_SHELL_CHARS = /^[%+,\-./:=@_0-9A-Za-z]+$/
+
+export function shellQuote (arg: string): string {
+  if (arg === '') {
+    return '\'\''
```
Comment (sorccu) [inline packages/cli/src/services/shell.ts]: Alright but not sure why you changed these. I thought it was more readable with backticks instead of this forest of backslashes.
